src/main/java/com/cricket/project/service/impl/TeamServiceImpl.java

```diff
@@ -0,0 +1,117 @@
+                .teamPlayersId(playersInTeam)
+                .totalWins(0)
+                .totalMatchesPlayed(0)
+                .build();
+
+        teamRepository.saveTeam(team);
+        return team;
+
+    }
+
+    public List<Integer> setPlayersInTeam()
+    {
+        List<Player> bowlers = playerRepository.findPlayersByPlayerType(PlayerType.valueOf("BOWLER"));
+        List<Player> batsmen = playerRepository.findPlayersByPlayerType(PlayerType.valueOf("BATSMAN"));
+        List<Player> allRounders = playerRepository.findPlayersByPlayerType(PlayerType.valueOf("ALLROUNDER"));
+        List<Integer> playersInTeam = new ArrayList<>();
+
+        int numberOfBowlers = 4;
+        int numberOfBatsman = 4;
+        int numberOfAllRounder = 3;
+
+        setPlayersInTeamByPlayerType(allRounders, playersInTeam, numberOfAllRounder);
+        setPlayersInTeamByPlayerType(batsmen, playersInTeam, numberOfBatsman);
+        setPlayersInTeamByPlayerType(bowlers, playersInTeam, numberOfBowlers);
+        return playersInTeam;
+    }
+
+    private void setPlayersInTeamByPlayerType(List<Player> players, List<Integer> playersInTeam, int numberOfPlayerType) {
+        for(Player player:players)
+        {
+            int allRounderId = player.getId();
+            if(!exist.containsKey(allRounderId) && numberOfPlayerType>0)
+            {
+                exist.put(allRounderId,1);
+                playersInTeam.add(allRounderId);
+                numberOfPlayerType--;
+            }
+        }
+    }
+
+    @Override
+    public Team addPlayerToTeam(int playerId, int teamId) throws TeamException {
+        Team team = teamRepository.findTeamById(teamId);
+        List<Integer> playerIds = team.getTeamPlayersId();
+        if(playerIds.contains(playerId))
+        {
+            throw new TeamException("Player Already In Team");
+        }
+        if(playerIds.size()==11)
+        {
+            throw new TeamException("Team Already Has 11 Players");
+        }
+        playerIds.add(playerId);
+        team.setTeamPlayersId(playerIds);
+        team.setTeamCaptain(playerRepository.findPlayerById(playerIds.get(0)).getPlayerName());
+        teamRepository.saveTeam(team);
+        return team;
+     }
+
+    @Override
+    public Team removePlayerFromTeam(int playerId, int teamId) throws TeamException {
+        Team team = teamRepository.findTeamById(teamId);
+        List<Integer> playerIds = team.getTeamPlayersId();
+        if (playerIds.isEmpty()) {
+            throw new TeamException("Team is empty");
+        }
+        if (!playerIds.contains(playerId)) {
+            throw new TeamException("Player not found in the team");
+        }
+        playerIds.removeIf(id -> id == playerId);
+        team.setTeamPlayersId(playerIds);
+        team.setTeamCaptain(playerRepository.findPlayerById(playerIds.get(0)).getPlayerName());
+        teamRepository.saveTeam(team);
```

Comment:
You should return the team that is saved in the repo not the team that we modified in the method, sometimes things might not be saved properly or some values are not set properly and we might get wrong idea from the response

Reply:
changes made.